### ISSUE TITLE
nu --ide-ast bugs

### ISSUE THREAD

**Issue by maxim-uvarov** [nushell/nushell#10542]
### Describe the bug

`nu --ide-ast` doesn't highlight some tokens properly
<img width="588" alt="image" src="https://github.com/nushell/nushell/assets/4896754/78bde7d2-ade8-4bc4-b8d0-518b33430526">

### How to reproduce

Create a .nu file with a content below and open it in vscode:

```
mkdir test; cd test # cd is not highlighted

def 'abc' [] {} # def and 'abc' are incorrect

def abcd [] {} # all is correct here
```

### Expected behavior

I expected nu to highlight `; cd`, and `def 'abc'` properly

### Configuration

| key                | value                                         |
| ------------------ | --------------------------------------------- |
| version            | 0.85.0                                        |
| branch             |                                               |
| commit_hash        |                                               |
| build_os           | macos-aarch64                                 |
| build_target       | aarch64-apple-darwin                          |
| rust_version       | rustc 1.70.0 (90c541806 2023-05-31)           |
| rust_channel       | stable-aarch64-apple-darwin                   |
| cargo_version      | cargo 1.70.0 (ec8a8a0ca 2023-04-25)           |
| build_time         | 2023-09-23 05:58:06 +00:00                    |
| build_rust_channel | release                                       |
| allocator          | mimalloc                                      |
| features           | dataframe, default, sqlite, trash, which, zip |
| installed_plugins  |                                               |

**Comment by fdncred** [comment]
I don't believe --ide-ast controls highlighting. It's more likely the regular expressions in the vscode extension, since it currently doesn't support semantic highlighting. I'll move this issue over to that repo.
